Tests/OptimizelyTests-DataModel/HoldoutConfigTests.swift

```diff
@@ -0,0 +1,184 @@
+        var included: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        included.id = "i1"
+        included.includedFlags = ["f"]
+        
+        var config = HoldoutConfig(allholdouts: [included, global1, global2])
+        
+        let result = config.getHoldoutForFlag(id: "f").map(\.id)
+        XCTAssertEqual(result, ["g1", "g2", "i1"])
+    }
+    
+    func testHoldoutOrdering_with_Both_IncludedAndExcludedFlags() {
+        let flag1 = "11111"
+        let flag2 = "22222"
+        let flag3 = "33333"
+        let flag4 = "44444"
+        
+        var inc: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        inc.id = "i1"
+        inc.includedFlags = [flag1]
+        
+        var exc: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        exc.id = "e1"
+        exc.excludedFlags = [flag2]
+        
+        var gh1: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        gh1.id = "gh1"
+        gh1.includedFlags = []
+        gh1.excludedFlags = []
+        
+        var gh2: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        gh2.id = "gh2"
+        gh2.includedFlags = []
+        gh2.excludedFlags = []
+        
+        
+        let allHoldouts =  [inc, exc, gh1, gh2]
+        var holdoutConfig = HoldoutConfig(allholdouts: allHoldouts)
+        
+        XCTAssertEqual(holdoutConfig.getHoldoutForFlag(id: flag1), [exc, gh1, gh2, inc])
+        XCTAssertEqual(holdoutConfig.getHoldoutForFlag(id: flag2), [gh1, gh2])
+        
+        XCTAssertEqual(holdoutConfig.getHoldoutForFlag(id: flag3), [exc, gh1, gh2])
+        XCTAssertEqual(holdoutConfig.getHoldoutForFlag(id: flag4), [exc, gh1, gh2])
+        
+    }
+    
+    func testExcludedHoldout_shouldNotAppearInGlobalForFlag() {
+        var global: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        global.id = "global"
+        
+        var excluded: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        excluded.id = "excluded"
+        excluded.excludedFlags = ["f"]
+        
+        var config = HoldoutConfig(allholdouts: [global, excluded])
+        
+        let result = config.getHoldoutForFlag(id: "f").map(\.id)
+        XCTAssertEqual(result, ["global"]) // excluded should not appear
+    }
+    
+    func testGetHoldoutForFlag_shouldUseCacheOnSecondCall() {
+        var holdout: Holdout = try! OTUtils.model(from: HoldoutTests.sampleData)
+        holdout.id = "1"
+        holdout.excludedFlags = ["f1"]
+        
+        var config = HoldoutConfig(allholdouts: [holdout])
+        
+        _ = config.getHoldoutForFlag(id: "f1")
+        let writeCountAfterFirst = config.cacheWriteCount
+        
+        _ = config.getHoldoutForFlag(id: "f1")
+        let writeCountAfterSecond = config.cacheWriteCount
+        
+        XCTAssertEqual(writeCountAfterFirst, 1)
+        XCTAssertEqual(writeCountAfterSecond, 1, "Second call should not increase cache write count")
```

Comment:
I understand mocking is not supported in swift, but odd with increasing count :) I hope other SDKs won't follow this.